src/middlewares.js

```diff
@@ -42,6 +42,42 @@ function cache(store, params) {
 }
 
 /**
+ * Leveraging HTTP cache control.
+ */
+function httpCache(params) {
+  return function(req, res, next) {
+    var header = null;
+
+    if (typeof params === 'string')
+      header = params;
+
+    // If params is an object, use special properties
+    // defined by API to set `max-age`.
+    if (params !== null && typeof params === 'function' || typeof params === 'object') {
+
+      var durations = {
+        seconds: 1, minutes: 60, hours: 3600, days: 86400, weeks: 604800
+      };
+
+      Object.keys(durations).some(function(key) {
+        if (params.hasOwnProperty(key)) {
+          header = 'private, max-age=' + durations[key] * params[key];
+          return header;
+        }
+      });
+    }
+
+    if (header) {
+      res.set('Cache-Control', header);
+    } else {
+      return next(new Error('Wrong parameter given for HTTP cache control'));
```

Comment:
@Yomguithereal Apart from implementation discussion... do you have a way to silence the exception thrown in console when running tests?